src/deploy_pex.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python
+
+# Switches PEX deploy behavior based on github runner's ubuntu version
+# ubuntu-20.04 can always build pexes that work on our target platform
+# ubuntu-22.04 can only build pexes if there are no sdists (source only packages)
+
+# On ubuntu-20.04: forward args to builder.pex
+
+# On ubuntu-22.04:
+#   Try to build without sdists
+#     If success: all good (no sdists were involved)
+#     If failure: run builder.pex within the dagster-manylinux-builder docker image
+
+import os
+from pathlib import Path
+import subprocess
+import sys
+
+BUILDER_PEX_PATH = Path(__file__).parent.parent / "generated/gha/builder.pex"
+DOCKER_PATH = "/usr/bin/docker"
+DOCKER_IMAGE = "ghcr.io/dagster-io/dagster-manylinux-builder:v0.1"
+
+
+def main():
+    args = sys.argv[1:]
+    ubuntu_version = get_runner_ubuntu_version()
+    print("Running on Ubuntu", ubuntu_version)
+    if ubuntu_version == "20.04":
+        returncode, output = deploy_pex_from_current_environment(args, build_sdists=True)
+    else:
+        returncode, output = deploy_pex_from_current_environment(args, build_sdists=False)
+        if returncode:
+            dep_failures = dependency_failure_lines(output)
+            if dep_failures:
+                try:
+                    print(
+                        "::group::Preparing a Docker build environment to build the PEX files",
+                        flush=True,
+                    )
+                    print(
+                        "Going to build dependencies within a Docker build environment "
+                        "to resolve missing binary packages for the following:"
+                    )
+                    for line in dep_failures:
+                        print(f"- {line}")
+
+                    returncode, output = deploy_pex_from_docker(args)
+                finally:
+                    print("::endgroup::", flush=True)
+
+    if returncode:
+        print(
+            "::error Title=Deploy failed::Failed to deploy Python Executable. "
+            "Try removing `ENABLE_FAST_DEPLOYS: 'true'` from your .github/workflows/*yml."
```

Comment:
maybe recommend setting it to 'false' instead so its easier to undo later?

Reply:
done